python/lsst/ts/hvac/mqtt_info_reader.py

```diff
@@ -124,318 +104,217 @@ def __init__(self) -> None:
+        m = GENERAL_REGEX.match(mqtt_topic)
+        if m:
+            return m.group(), None
+        raise KeyError(f"No known device found in {mqtt_topic}.")
+
+    def _validate_all_rows_contain_know_devices(
+        self, mqtt_topic_rows: list[dict[str, str]]
+    ) -> None:
+        devices: set[str] = set()
+        for row in mqtt_topic_rows:
+            topic = row["TOPIC MQTT"]
+            device, _ = self._extract_device(topic)
+            devices.add(device)
+
+        for device in devices:
+            HvacTopicEnglish(device)
+        for hvac_device in HvacTopicEnglish:
+            assert hvac_device.value in devices, f"{hvac_device.value} is missing."
+
+    def _validate_all_event_topics(
+        self, event_topic_rows: list[dict[str, str]]
+    ) -> None:
+        all_event_topic_rows = [row["TOPIC MQTT"] for row in event_topic_rows]
+        for event_topic in EVENT_TOPICS:
+            assert (
+                event_topic in all_event_topic_rows
+            ), f"{event_topic} doesn't exist in event_topic_rows."
+        for event_topic in all_event_topic_rows:
+            assert (
+                event_topic in EVENT_TOPICS
+            ), f"{event_topic} doesn't exist in EVENT_TOPICS."
+
+    def _collect_devices_and_items(
+        self, topic_rows: list[dict[str, str]], sal_topic_type: SalTopicType
+    ) -> None:
+        for row in topic_rows:
+            hvac_topic_and_item = row["TOPIC MQTT"]
+            idl_type = "float" if "ANALOG" in row["SIGNAL"] else "boolean"
+            topic_type = row["READ / WRITE"].strip()
+            if topic_type in [TopicType.READ, TopicType.WRITE]:
+                unit = determine_unit(row["UNIT"])
+                limits = parse_limits(row["LIMITES"].strip())
+            else:
+                raise KeyError(f"Found unknown topic type {topic_type}.")
+
+            device, item, _ = self.extract_topic_and_item(hvac_topic_and_item)
+            hvac_topic = HvacTopicEnglish(device)
+            hvac_item: TelemetryItemEnglish | CommandItemEnglish | EventItem = (
+                TelemetryItemEnglish.alarmDevice
             )
-            for index, row in csv_reader.iterrows():
-                csv_hvac_topic_and_item = row["topic_and_item"]
-                idl_type = "float" if "ANALOG" in row["signal"] else "boolean"
-                topic_type = row["rw"].strip()
-                if topic_type in [TopicType.READ, TopicType.WRITE]:
-                    unit = self._determine_unit(row["unit"])
-                    limits = self._parse_limits(row["limits"].strip())
-                    csv_hvac_topics[csv_hvac_topic_and_item] = {
-                        "idl_type": idl_type,
-                        "topic_type": topic_type,
-                        "unit": unit,
-                        "limits": limits,
-                    }
-        self._collect_topics_and_items(csv_hvac_topics)
+            hvac_topics = self.telemetry_topics
```

Comment:
I think this default value won't be ever used as the following match-case block enforces to take a value (or raises an error otherwise). Can you confirm :pray:?

Reply:
I wonder if the same happens for the above default value of hvac_item:
```python
            hvac_item: TelemetryItemEnglish | CommandItemEnglish | EventItem = (
                TelemetryItemEnglish.alarmDevice
            )
```

Reply:
You are right. I added that line to resolve a warning in my IDE.